src/Microsoft.Performance.SDK.Tests/TableDiscoveryTests.cs

```diff
@@ -0,0 +1,170 @@
+    {
+        [TestMethod]
+        [UnitTest]
+        public void NoTablesInAssemblyLeavesEmptyProperties()
+        {
+            var assembly = new FakeAssembly
+            {
+                TypesToReturn = new[]
+                {
+                    typeof(DateTime),
+                }
+            };
+
+            var sut = TableDiscovery.CreateForAssembly(assembly);
+            var tables = sut.Discover(new FakeSerializer());
+
+            Assert.AreEqual(0, tables.Count);
+        }
+
+        [TestMethod]
+        [UnitTest]
+        public void OnlyDataTablesInAssemblyPopulatesCorrectly()
+        {
+            var assembly = new FakeAssembly
+            {
+                TypesToReturn = new[]
+                {
+                    typeof(DateTime),
+                    typeof(StubDataTableOne),
+                    typeof(StubDataTableTwo),
+                    typeof(StubDataTableThree),
+                    typeof(Exception),
+                }
+            };
+
+            var serializer = new FakeSerializer();
+
+            Assert.IsTrue(
+                TableDescriptorFactory.TryCreate(
+                    typeof(StubDataTableOne),
+                    serializer,
+                    out TableDescriptor expectedDescriptor1));
+            Assert.IsTrue(
+                TableDescriptorFactory.TryCreate(
+                    typeof(StubDataTableTwo),
+                    serializer,
+                    out TableDescriptor expectedDescriptor2));
+            Assert.IsTrue(
+                TableDescriptorFactory.TryCreate(
+                    typeof(StubDataTableThree),
+                    serializer,
+                    out TableDescriptor expectedDescriptor3));
+
+            var sut = TableDiscovery.CreateForAssembly(assembly);
+            var tables = sut.Discover(new FakeSerializer());
+
+            Assert.AreEqual(3, tables.Count);
+            Assert.IsNotNull(tables.Single(x => x.Descriptor.Equals(expectedDescriptor1)));
+            Assert.IsNotNull(tables.Single(x => x.Descriptor.Equals(expectedDescriptor2)));
+            Assert.IsNotNull(tables.Single(x => x.Descriptor.Equals(expectedDescriptor3)));
+        }
+
+        [TestMethod]
+        [UnitTest]
+        public void OnlyMetadataTablesInAssemblyPopulatesCorrectly()
+        {
+            var assembly = new FakeAssembly
+            {
+                TypesToReturn = new[]
+                {
+                    typeof(DateTime),
+                    typeof(StubMetadataTableOne),
+                    typeof(StubMetadataTableTwo),
+                    typeof(Exception),
+                }
+            };
+
+            var serializer = new FakeSerializer();
+
+            Assert.IsTrue(
+                TableDescriptorFactory.TryCreate(
+                    typeof(StubMetadataTableOne),
+                    serializer,
+                    out TableDescriptor expectedDescriptor1));
+            Assert.IsTrue(
```

Comment:
notice this is done a lot. Maybe we need a test util file that takes in a list of types and has out param for a list of descriptors?

Reply:
That makes sense, thanks.